sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JsonInferSchema.scala

```diff
@@ -68,13 +71,19 @@ private[sql] class JsonInferSchema(options: JSONOptions) extends Serializable {
             Some(inferField(parser))
           }
         } catch {
-          case  e @ (_: RuntimeException | _: JsonProcessingException) => parseMode match {
-            case PermissiveMode =>
-              Some(StructType(Seq(StructField(columnNameOfCorruptRecord, StringType))))
-            case DropMalformedMode =>
+          case e @ (_: RuntimeException | _: JsonProcessingException | _: IOException) =>
+            if (ignoreCorruptFiles) {
+              logWarning(s"Skipped the corrupted file: $row", e)
```

Comment:
Can we keep the log consistent with other places? e.g.) Skipped the rest of the content in ...

Reply:
here we use it for inferring schema, not all the content.

At avro side `org.apache.spark.sql.avro.AvroUtils#inferAvroSchemaFromFiles`, we say
```
Skipped the footer in the corrupted file: ...
```

How about we print `Skipped inferring schema from the corrupted file: ...`

Reply:
Also should we maybe exclude `RuntimeException` for now? I think we intentionally throw `RuntimeException` for some places like:

```
    case token =>
      // We cannot parse this token based on the given data type. So, we throw a
      // RuntimeException and this exception will be caught by `parse` method.
      throw QueryExecutionErrors.failToParseValueForDataTypeError(parser, token, dataType)
```

Reply:
That's fine. but just to make extra sure `row` is being printed, not the file name. Maybe we should handle it in a higher level?

Reply:
seems very hard to do it at a higher level